src/ast/values.ts

```diff
@@ -257,31 +271,28 @@ const callsArgReturnsNumber: RawMemberDescription = {
 	}
 };
 
-const UNKNOWN_LITERAL_STRING: ExpressionEntity = {
-	deoptimizePath: () => {},
-	getLiteralValueAtPath: () => UnknownValue,
-	getReturnExpressionWhenCalledAtPath: path => {
+const UNKNOWN_LITERAL_STRING: ExpressionEntity = new class extends ValueBase {
+	getReturnExpressionWhenCalledAtPath(path: ObjectPath) {
 		if (path.length === 1) {
 			return getMemberReturnExpressionWhenCalled(literalStringMembers, path[0]);
 		}
 		return UNKNOWN_EXPRESSION;
-	},
-	hasEffectsWhenAccessedAtPath: path => path.length > 1,
-	hasEffectsWhenAssignedAtPath: path => path.length > 0,
-	hasEffectsWhenCalledAtPath: (path, callOptions, context) => {
+	}
+	hasEffectsWhenAccessedAtPath(path: ObjectPath) { return path.length > 1 }
+	hasEffectsWhenCalledAtPath(
+		path: ObjectPath,
+		callOptions: CallOptions,
+		context: HasEffectsContext
+	) {
 		if (path.length === 1) {
 			return hasMemberEffectWhenCalled(literalStringMembers, path[0], true, callOptions, context);
 		}
 		return true;
-	},
-	include: () => {},
+	}
 	includeCallArguments(context: InclusionContext, args: (ExpressionNode | SpreadElement)[]): void {
-		for (const arg of args) {
-			arg.include(context, false);
-		}
-	},
-	included: true,
-	toString: () => '[[UNKNOWN STRING]]'
+		includeAll(context, args);
+	}
+	toString() { return '[[UNKNOWN STRING]]' }
```

Comment:
Actually these "toString" are only here for debugging and I do not think they turned out to be any useful. We should just remove them all.